From cee6817c2f4637f859829c7caeac168ccbb74850 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Sun, 23 Dec 2018 13:29:27 +0100
Subject: [PATCH] Also test memcheck/tests/vbit-test on any secondary arch.

If we are building a secondary arch then also build and run ther
memcheck vbit-test for that architecture.
---
 memcheck/tests/vbit-test/Makefile.am              | 20 ++++++++++++++++++++
 memcheck/tests/vbit-test/vbit-test-sec.stderr.exp |  0
 memcheck/tests/vbit-test/vbit-test-sec.vgtest     |  3 +++
 3 files changed, 23 insertions(+)
 create mode 100644 memcheck/tests/vbit-test/vbit-test-sec.stderr.exp
 create mode 100644 memcheck/tests/vbit-test/vbit-test-sec.vgtest

diff --git a/memcheck/tests/vbit-test/Makefile.am b/memcheck/tests/vbit-test/Makefile.am
index 371a71b..124e82e 100644
--- a/memcheck/tests/vbit-test/Makefile.am
+++ b/memcheck/tests/vbit-test/Makefile.am
@@ -17,6 +17,10 @@ noinst_HEADERS = vtest.h vbits.h
 
 noinst_PROGRAMS = vbit-test
 
+if VGCONF_HAVE_PLATFORM_SEC
+noinst_PROGRAMS += vbit-test-sec
+endif
+
 if VGCONF_OS_IS_DARWIN
 noinst_DSYMS = $(noinst_PROGRAMS)
 endif
@@ -41,3 +45,19 @@ vbit_test_CFLAGS       = $(AM_CFLAGS_PRI)
 vbit_test_DEPENDENCIES = 
 vbit_test_LDADD        = $(top_builddir)/VEX/libvex-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 vbit_test_LDFLAGS      = $(AM_CFLAGS_PRI) @LIB_UBSAN@
+
+if VGCONF_HAVE_PLATFORM_SEC
+vbit_test_sec_SOURCES      = $(SOURCES)
+vbit_test_sec_CPPFLAGS     = $(AM_CPPFLAGS_SEC) \
+                             $(AM_CPPFLAGS_@VGCONF_PLATFORM_SEC_CAPS@) \
+                             -I$(top_srcdir)/include  \
+                             -I$(top_srcdir)/memcheck \
+                             -I$(top_srcdir)/VEX/pub
+vbit_test_sec_CFLAGS       = $(AM_CFLAGS_SEC) \
+                             $(AM_CFLAGS_@VGCONF_PLATFORM_SEC_CAPS@)
+vbit_test_sec_DEPENDENCIES = 
+vbit_test_sec_LDADD        = $(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
+                              $(TOOL_LDADD_@VGCONF_PLATFORM_SEC_CAPS@)
+vbit_test_sec_LDFLAGS      = $(AM_CFLAGS_SEC) @LIB_UBSAN@ \
+                             $(TOOL_LDFLAGS_@VGCONF_PLATFORM_SEC_CAPS@)
+endif
diff --git a/memcheck/tests/vbit-test/vbit-test-sec.stderr.exp b/memcheck/tests/vbit-test/vbit-test-sec.stderr.exp
new file mode 100644
index 0000000..e69de29
diff --git a/memcheck/tests/vbit-test/vbit-test-sec.vgtest b/memcheck/tests/vbit-test/vbit-test-sec.vgtest
new file mode 100644
index 0000000..2d3c938
--- /dev/null
+++ b/memcheck/tests/vbit-test/vbit-test-sec.vgtest
@@ -0,0 +1,3 @@
+prog: vbit-test-sec
+prereq: test -x vbit-test-sec
+vgopts: -q --expensive-definedness-checks=yes
-- 
1.8.3.1
